test/server/cards/02_SHD/leaders/FinnThisIsARescue.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('Finn, This is a Rescue', function () {
+    integration(function (contextRef) {
+        describe('Finn\'s undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['jedi-lightsaber'],
+                        leader: { card: 'finn#this-is-a-rescue', deployed: false },
+                        groundArena: ['battlefield-marine'],
```

Comment:
You can have this as:
`groundArena: [ { card: 'battlefield-marine', upgrades: [ 'jedi-lightsaber' ] } ]`